controllers/components/teams/team.jsx

```diff
@@ -0,0 +1,218 @@
+                                </li>
+                            </ul>
+                        </div>
+                    );
+                } else {
+                    return '';
+                }
+            });
+
+            let companyName = this.props.company;
+            let title = players.length ? 'All Players' : 'This team has ZERO players';
+
+            return (
+                <div className = 'container team-content'>
+                    <div className = 'row justify-content-md-center'>
+                        <div className = 'col-md-auto'>
+                            <h3>Admin Dashboard</h3>
+                        </div>
+                    </div>
+                    <div className = 'row'>
+                        <div className = 'col col-sm-4'>
+                            <div className = 'team-info'>
+                                <h4>Team Name: <span>{ teamName }</span></h4>
+                                <h4>Company: <span>{ companyName }</span></h4>
+                            </div>
+                            <button className = 'btn btn-danger'>Delete Team</button>
+                        </div>
+                        <div className = 'players-wrap col col-sm-7'>
+                            <p>{ title }</p>
+                            { players }
+                        </div>
+                    </div>
+                    <div className = 'row'>
+                        <div className = 'col col-sm-4'>
+                        </div>
+                    </div>
+                </div>
+            );
+
+        } else {
+            return <TeamNotFound />;
+        }
+    }
+
+}
+
+class TeamMempers extends React.Component {
+    constructor( props ) {
+        super( props );
+
+        this.state = {
+            players: [],
+            company: '',
+            isValidTeam: 'no',
+            pending: ''
+        }
+    }
+
+    componentWillMount() {
+        const that = this;
+
+        // get players of the team
+        fetch( '/api/search/team/' + teamName + '/players')
+        .then( ( res ) => {
+            if ( !res.ok ) return errorHandler( res );
+            res.json().then( ( data ) => that.setState({
+                isValidTeam: true,
+                players:data[ 0 ],
+                company: data[ 1 ]
+            }) )
+        }).catch( ( err) => alert( err ) );
+
+        function errorHandler( err, message ) {
+            if ( err.status >= 400 && err.status < 500 )
+                return err.text().then( ( message) => { if ( message == 'no players'){ that.setState({ isValidTeam: true })} } );
+            if ( message ) return alert( message );
+            return alert( 'There was a problem submitting your form. Please try again later' );
+        }
+    }
+
+    _identifyUser() {
+        if ( localStorage.token ) {
+            const user = JSON.parse( atob( localStorage.token.split( '.' )[ 1 ] ) );

```

Comment:
Stop doing that all over, do it via a helper function somewhere.

Reply:
I put it as a helper function beside regex in `helpers.js`, will use it while changing most of components because i changed the payload in token.

Reply:
Thank you